internal/importer/importer.go

```
@@ -0,0 +1,250 @@
+	}
+	return plan, nil
+}
+
+// PlanURL fetches a button spec from a URL. Supports self-contained HTTP/API
+// button specs (the kind `create --url` produces); code/prompt buttons need
+// their bundle, so use `buttons install` (the registry) for those.
+func PlanURL(url, nameOverride string) (*Plan, error) {
+	if !strings.HasPrefix(url, "http://") && !strings.HasPrefix(url, "https://") {
+		return nil, fmt.Errorf("url must start with http:// or https://")
+	}
+	resp, err := http.Get(url) // #nosec G107 -- user-supplied import URL, fetched read-only and size-capped
```

golangci-lint (2.12.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)